tests/components/sun/test_init.py

```diff
@@ -152,10 +155,10 @@ async def test_norway_in_june(hass):
 
     assert dt_util.parse_datetime(
         state.attributes[sun.STATE_ATTR_NEXT_RISING]
-    ) == datetime(2016, 7, 25, 23, 23, 39, tzinfo=dt_util.UTC)
+    ) == datetime(2016, 7, 24, 22, 59, 45, 689645, tzinfo=dt_util.UTC)
```

Comment:
Why does this change?

Reply:
(copy-paste of relevant information from above)
The new Astral both produces different times and provides times down to the MS level. To deal with that the tests need to be re-written. Tests were written to match old Astrals less accurate data.

I got confirmation here:
https://github.com/home-assistant/core/pull/48282#issuecomment-808855672

Reply:
The difference looks to be more than 24 hours. So it shouldn't just be the milliseconds accuracy. Can you explain the other differences that would result in this change?

Reply:
Good question, this confused and stumped me for some time. Long story short, since the day goes on for more than a month, it seems many calculators are in disagreement as to the date of sunrise and sunset here.

For example:
https://www.timeanddate.com/sun/norway/tromso?month=6&year=2016
https://www.esrl.noaa.gov/gmd/grad/solcalc/table.php?lat=69.6&lon=18.8&year=2016

It seems both versions of astral are wrong, but is entirely dependant on the altitude specified, as the sun barely makes it to the 7 degree inclination required for it to be considered a sunrise, the more altitude you add the sooner time the sunrise is ( in this case even by almost a day! ) The new version of Astral uses the location of the user ( via hass config ) to calculate times, as such the sunrise and sunset times will vary because of that. You will see I made edits to the tests in many places to specify a location's elevation ( based off the elevation at that latitude and longitude of coordinates ) as to increase the accuracy of apparent sunrise.